microk8s-resources/actions/enable.host-access.sh

```diff
@@ -30,7 +30,7 @@ fi
 echo "$IP_ADDRESS" > "${SNAP_DATA}/var/lock/host-access-enabled"
 echo "Setting ${IP_ADDRESS} as host-access"
 
-run_with_sudo "$SNAP/sbin/ifconfig" lo:microk8s "$IP_ADDRESS" up
+run_with_sudo "$SNAP/sbin/ip" addr add "$IP_ADDRESS"/32 dev lo label lo:microk8s
```

Comment:
`ifconfig lo:microk8s 10.0.1.1 up` assigns the IP address 10.0.1.1/**8** to the interface.

This is all link-local, so the subnet size does not matter.

Reply:
I would prefer to preserve the existing behaviour, but this is fine as well.

Reply:
About the /8, I believe it is a behavior inherited from ifconfig as an IP from between 0.X.X.X and 127.X.X.X was considered an A class subnet coming in /8 subnet sizes and without any details ifconfig would configure it like this.
If for example the user decides to specify an IP like 192.168.243.1, ifconfig would then suppose it is a class C address and then add the IP address 192.168.243.1/24 but not a /8 so the behavior is not even consistent as it is today.

On the opposite, **ip addr** would create by default a /32 even if nothing is precised after the IP address.
I only added the /32 to be consistent to the command when disabling the host-access and thus removing the IP on localhost interface and to make sure the behavior stays exactly the same even if **ip addr** changes its default subnet size in the future.

When deleting an ip without the subnet indicated, it would works but you would be greated with this message : 
> $ sudo ip addr del 10.0.1.1 dev lo label lo:microk8s
> Warning: Executing wildcard deletion to stay compatible with old scripts.
>          Explicitly specify the prefix length (10.0.1.1/32) to avoid this warning.
>          This special behaviour is likely to disappear in further releases,
>          fix your scripts!

As I don't have perfect knowledge of microk8s' networking, if required the subnet can be expended to a less specific CIDR, however it shouldn't be a /8 as this won't work with the remaining rfc1918 subnets, the behavior of Class A/B/C is something that is deprecated since a while and shouldn't need to be reproduced.